idea/testData/quickfix/expressions/removeUselessIsCheckInWhen.kt

```diff
@@ -0,0 +1,6 @@
+// "Remove useless is check" "false"
+fun foo(a: String) {
+    when (1) {
+        <caret>is Int -> { }
```

Comment:
I recommend never delete the whole when. For your examples, we should get

```
// false condition
when (1) {
}

// true condition
when (1) {
    else -> print()
}
```

I don't think it's something dangerous because user can always undo the optimization.

Keep in mind also when without subject case. Here we can just replace conditions into `true` and `false`, as for simple conditions.